integration/container/checkpoint_test.go

```diff
@@ -0,0 +1,171 @@
+package container // import "github.com/docker/docker/integration/container"
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	containertypes "github.com/docker/docker/api/types/container"
+	mounttypes "github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/internal/test/daemon"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+	"gotest.tools/poll"
+	"gotest.tools/skip"
+)
+
+func containerExec(t *testing.T, client *client.Client, cID string, cmd []string) {
+	ctx := context.Background()
+	id, err := client.ContainerExecCreate(ctx, cID,
+		types.ExecConfig{
+			WorkingDir: "/tmp",
+			Cmd:        cmd,
+		},
+	)
+	assert.NilError(t, err)
+
+	resp, err := client.ContainerExecAttach(ctx, id.ID,
+		types.ExecStartCheck{
+			Detach: false,
+			Tty:    false,
+		},
+	)
+	assert.NilError(t, err)
+	defer resp.Close()
+	// get the exit code
+	iresp, err := client.ContainerExecInspect(ctx, id.ID)
+	assert.NilError(t, err)
+	assert.Equal(t, iresp.ExitCode, 0)
+}
+
+func TestCheckpoint(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows" && testEnv.DaemonInfo.Isolation == "process")
+
+	defer setupTest(t)()
+
+	cmd := exec.Command("criu", "check")
+	stdoutStderr, err := cmd.CombinedOutput()
+	t.Logf("%s", stdoutStderr)
+	assert.NilError(t, err)
+
+	d := daemon.New(t, daemon.WithExperimental)
+	d.StartWithBusybox(t)
+	defer d.Stop(t)
+
+	client, err := d.NewClient()
+	assert.Check(t, err, "error creating client")
+	ctx := context.Background()
+
+	cfg := containertypes.Config{
+		Image: "busybox",
+		Cmd:   []string{"sh", "-c", "while :; do sleep 1; done"},
+	}
+	hostCfg := containertypes.HostConfig{
+		Mounts: []mounttypes.Mount{
+			{
+				Type:   mounttypes.TypeTmpfs,
+				Target: "/tmp",
+			},
+		},
+	}
+
+	resp, err := client.ContainerCreate(ctx, &cfg, &hostCfg, nil, "")
```

Comment:
I'd use `container.Run()` from integration/internal/container/container.go. In your case that'd be something like

```go
...
mnt := mounttypes.Mount{
￼  				Type:   mounttypes.TypeTmpfs,
￼  				Target: "/tmp",
￼}
cID := container.Run(t, ctx, client, container.WithMount(mnt))
 	poll.WaitOn(t,
￼  		container.IsInState(ctx, client, cID, "running"),
￼  		poll.WithDelay(100*time.Millisecond),
￼  	)
...
```